File: library/src/main/java/com/afollestad/aesthetic/AestheticSnackBarTextView.java
package com.afollestad.aesthetic;

import android.content.Context;
import android.support.v7.widget.AppCompatTextView;
import android.util.AttributeSet;

import rx.Subscription;

import static com.afollestad.aesthetic.Rx.onErrorLogAndRethrow;

/** @author Aidan Follestad (afollestad) */
final class AestheticSnackBarTextView extends AppCompatTextView {

  private Subscription subscription;

  public AestheticSnackBarTextView(Context context) {
    super(context);
  }

  public AestheticSnackBarTextView(Context context, AttributeSet attrs) {
    super(context, attrs);
  }

  public AestheticSnackBarTextView(Context context, AttributeSet attrs, int defStyleAttr) {
    super(context, attrs, defStyleAttr);
  }

  @Override
  protected void onAttachedToWindow() {
    super.onAttachedToWindow();
    subscription =
        Aesthetic.get()
            .snackbarTextColor()
            .compose(Rx.<Integer>distinctToMainThread())
            .subscribe(ViewTextColorAction.create(this), onErrorLogAndRethrow());
  }

  @Override
  protected void onDetachedFromWindow() {
    subscription.unsubscribe();
    super.onDetachedFromWindow();
  }
}

File: library/src/main/java/com/afollestad/aesthetic/AestheticSnackBarTextView.kt
package com.afollestad.aesthetic

import android.content.Context
import android.support.v7.widget.AppCompatTextView
import android.util.AttributeSet
import com.afollestad.aesthetic.utils.distinctToMainThread
import com.afollestad.aesthetic.utils.onErrorLogAndRethrow
import io.reactivex.disposables.Disposable

/** @author Aidan Follestad (afollestad)
 */
internal class AestheticSnackBarTextView(
  context: Context,
  attrs: AttributeSet? = null
) : AppCompatTextView(context, attrs) {

  private var subscription: Disposable? = null

  override fun onAttachedToWindow() {
    super.onAttachedToWindow()
    subscription = Aesthetic.get()
        .snackbarTextColor()
        .distinctToMainThread()
        .subscribe(ViewTextColorAction(this), onErrorLogAndRethrow())
  }

  override fun onDetachedFromWindow() {
    subscription?.dispose()
    super.onDetachedFromWindow()
  }
}
